prow/pubsub/reporter/reporter.go

```diff
@@ -72,7 +72,16 @@ func (c *Client) GetName() string {
 
 // ShouldReport tells if a prowjob should be reported by this reporter
 func (c *Client) ShouldReport(pj *kube.ProwJob) bool {
-	return pj.Labels[PubSubProjectLabel] != "" && pj.Labels[PubSubTopicLabel] != ""
+	// Support checking for both labels(deprecated) and annotations(new) for backward compatibility
+	for _, label := range []string{PubSubProjectLabel, PubSubTopicLabel} {
```

Comment:
Since you do it a couple times, might be nice to DRY it out

Reply:
Done